### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by Hardexit** [PSAppDeployToolkit/PSAppDeployToolkit#1200]
### Prerequisites

- [X] Ensure you write a short, descriptive title after [Bug] above.
- [X] Make sure to [search for any existing issues](https://github.com/psappdeploytoolkit/psappdeploytoolkit/issues) before filing a new one.
- [X] Verify you are able to reproduce the issue with the [latest released version](https://www.github.com/psappdeploytoolkit/psappdeploytoolkit/releases/latest)

### PSAppDeployToolkit version

4.0.3

### Describe the bug

Whenever I call the script [Invoke-ServiceUI.ps1](https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/blob/fb0b22abfc4eb256538d8f89cb95d281f75cd6a6/examples/ServiceUI/Invoke-ServiceUI.ps1), I get an exitcode 1

### Steps to reproduce

```
.\Invoke-ServiceUI.ps1 -DeploymentType 'Install' -AllowRebootPassThru
DEBUG:    1+  >>>> .\Invoke-ServiceUI.ps1 -DeploymentType 'Install' -AllowRebootPassThru
DEBUG:     ! CALL function '<ScriptBlock>'
DEBUG:   27+     [string[]]$ProcessName =  >>>> @('explorer'),
DEBUG:   35+  >>>> Push-Location $PSScriptRoot
DEBUG:     ! CALL function '<ScriptBlock>'  (defined in file 'C:\Temp\test\Invoke-ServiceUI.ps1')
DEBUG:   37+ if ( >>>> $env:PROCESSOR_ARCHITECTURE -eq 'AMD64' -or $env:PROCESSOR_ARCHITEW6432 -eq 'AMD64') {
DEBUG:   38+      >>>> $Architecture = 'x64'
DEBUG:     ! SET $Architecture = 'x64'.
DEBUG:   43+ if ( >>>> Get-Process -Name $ProcessName -ErrorAction SilentlyContinue) {
DEBUG:   43+ if ( >>>> Get-Process -Name $ProcessName -ErrorAction SilentlyContinue) {
DEBUG:   44+     if ( >>>> [Environment]::UserInteractive) {
DEBUG:   46+          >>>> $Process = Start-Process -FilePath '.\Invoke-AppDeployToolkit.exe' -ArgumentList
"-DeploymentType $DeploymentType -DeployMode Interactive -AllowRebootPassThru:`$$AllowRebootPassThru
-TerminalServerMode:`$$TerminalServerMode -DisableLogging:`$$DisableLogging" -NoNewWindow -Wait -PassThru
DEBUG:   19+                                         if ( &  >>>> { Set-StrictMode -Version 1; $_.PSMessageDetails } )
{
DEBUG:     ! CALL function '<ScriptBlock>'
DEBUG:   19+                                         if ( & {  >>>> Set-StrictMode -Version 1; $_.PSMessageDetails } )
{
DEBUG:   19+                                         if ( & { Set-StrictMode -Version 1;  >>>> $_.PSMessageDetails } )
{
DEBUG:    1+ &  >>>> { Set-StrictMode -Version 1; $this.Exception.InnerException.PSMessageDetails }
DEBUG:     ! CALL function '<ScriptBlock>'
DEBUG:    1+ & {  >>>> Set-StrictMode -Version 1; $this.Exception.InnerException.PSMessageDetails }
DEBUG:    1+ & { Set-StrictMode -Version 1;  >>>> $this.Exception.InnerException.PSMessageDetails }
DEBUG:    1+ & { Set-StrictMode -Version 1; $this.Exception.InnerException.PSMessageDetails  >>>> }
DEBUG:   19+                                         if ( & { Set-StrictMode -Version 1; $_.PSMessageDetails  >>>> } )
{
DEBUG:   26+                                         $errorCategoryMsg = &  >>>> { Set-StrictMode -Version 1;
$_.ErrorCategory_Message }
DEBUG:     ! CALL function '<ScriptBlock>'
DEBUG:   26+                                         $errorCategoryMsg = & {  >>>> Set-StrictMode -Version 1;
$_.ErrorCategory_Message }
DEBUG:   26+                                         $errorCategoryMsg = & { Set-StrictMode -Version 1;  >>>>
$_.ErrorCategory_Message }
DEBUG:   26+                                         $errorCategoryMsg = & { Set-StrictMode -Version 1;
$_.ErrorCategory_Message  >>>> }
DEBUG:   42+                                         $originInfo = &  >>>> { Set-StrictMode -Version 1; $_.OriginInfo
}
DEBUG:     ! CALL function '<ScriptBlock>'
DEBUG:   42+                                         $originInfo = & {  >>>> Set-StrictMode -Version 1; $_.OriginInfo
}
DEBUG:   42+                                         $originInfo = & { Set-StrictMode -Version 1;  >>>> $_.OriginInfo
}
DEBUG:   42+                                         $originInfo = & { Set-StrictMode -Version 1; $_.OriginInfo  >>>>
}
Der Wert kann nicht in den Typ "System.String" konvertiert werden.
In C:\Temp\test\Invoke-ServiceUI.ps1:46 Zeichen:9
+         $Process = Start-Process -FilePath '.\Invoke-AppDeployToolkit ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvalidCastFromAnyTypeToString

DEBUG:   47+          >>>> $ExitCode = $Process.ExitCode
DEBUG:     ! SET $ExitCode = ''.
DEBUG:   58+  >>>> Pop-Location
DEBUG:   59+  >>>> exit $ExitCode
```

### Environment data

```console
OsName               : Microsoft Windows 11 Enterprise
OSDisplayVersion     : 21H2
OsOperatingSystemSKU : EnterpriseEdition
OsArchitecture       : 64-Bit
WindowsVersion       : 2009
WindowsProductName   : Windows 10 Enterprise
WindowsBuildLabEx    : 22000.1.amd64fre.co_release.210604-1628
OsLanguage           : de-DE
OsMuiLanguages       : {de-DE}
KeyboardLayout       : de-DE
TimeZone             : (UTC+01:00) Amsterdam, Berlin, Bern, Rom, Stockholm, Wien
HyperVisorPresent    : True
CsPartOfDomain       : True
CsPCSystemType       : Desktop

.NET SDK:
 Version:           9.0.101
 Commit:            eedb237549
 Workload version:  9.0.100-manifests.3068a692
 MSBuild version:   17.12.12+1cce77968

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\9.0.101\

Installierte .NET-Workloads:
Es sind keine installierten Workloads zum Anzeigen vorhanden.
Konfiguriert für die Verwendung loose manifests beim Installieren neuer Manifeste.

Host:
  Version:      9.0.0
  Architecture: x64
  Commit:       9d5a6a9aa4

.NET SDKs installed:
  9.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

**Comment by mjr4077au** [comment]
See my comment here: https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/discussions/1191#discussioncomment-11616072

> @Hardexit I can't see your comment regarding "Der Wert kann nicht in den Typ "System.String" konvertiert werden.", but please use these examples as they contain crucial fixes for v4: https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/tree/develop/examples/ServiceUI

**Comment by DanGough** [comment]
@JonathanPitre if you weren't testing with the dev branch copy, can you retry with the version that's now included in 4.0.4?
